Pex3

MRLGAnker opened the pull request:
Please make sure you are raising this PR against your own repository and not the original. If it says https://github.com/CorndelWithSoftwire in the search bar right now, you are in the wrong place!

Please also don't forget to submit the PR's url to Aptem afterwards (by clicking on the corresponding exercise component, pasting the url in the textbox that appears and then pressing the finish button).

MRLGAnker committed:
Completed Step 1:
    • Added View Model
    • Integrated View Model

MRLGAnker committed:
initial tests

MRLGAnker committed:
Completed Exercise 3

MRLGAnker committed:
Missed some files in previous commit

BenRamchandani commented on tests/test_todo.py:
```diff
@@ -0,0 +1,77 @@
+    @patch('requests.get')
+    def test_index_page(mock_get_requests, client):
+        mock_get_requests.side_effect = mock_request
+        response = client.get('/')
```
You should add some assertions here, e.g. check for a 200 response code and that your test data is included in the response.

BenRamchandani commented on tests_e2e/test_e2e.py:
```diff
@@ -0,0 +1,70 @@
+	try:
+		driver.find_element_by_xpath("//*[starts-with(@id,'to_do_item_')]")
+	except NoSuchElementException:
+		assert True
```
I'm not sure this tests anything, I think you want something like
```python
	with pytest.raises(NoSuchElementException):
		driver.find_element_by_xpath("//*[starts-with(@id,'to_do_item_')]")
```

BenRamchandani commented on tests_e2e/test_e2e.py:
```diff
@@ -0,0 +1,70 @@
+import pytest
+import os
+import selenium
```
Unused import

BenRamchandani commented on todo_app/data/session_items.py:
```diff
@@ -21,6 +22,53 @@ def __init__(self, id, name, idList, desc, due):
+        if len(self.done_items) < 5:
+            return True
+        else:
+            return False
```
`return len(self.done_items) < 5` 🙂

BenRamchandani commented on .env.test:
```diff
@@ -0,0 +1,8 @@
+# Change the following values for local development.
+SECRET_KEY=12345
+SECRET_TOKEN=1234567890
+BOARD_ID=B04RD
```
Can you add these too `.env.template` too?

BenRamchandani reviewed (changes requested):
Hi,

Your app and tests generally look good, the only change required to pass is to have at least one assertion in your integration test and mention how to run them in the README.

I've left a few other comments as well for you to look at if you've got time.

MRLGAnker committed:
- Made change recommended by Ben
- Removed Class definition in test_todo.py as it was breaking the tests
- Added asserts to test_todo.py so it actually tested the Mock
- Linted files
- Updated README.md with how to run tests
